From a8f18e4ba9b7deeb0d2912939e57ef6591e536a8 Mon Sep 17 00:00:00 2001
From: Marian-Cristian Rotariu <marian.rotariu@freescale.com>
Date: Wed, 29 Oct 2014 16:50:20 +0200
Subject: [PATCH 888/987] dpaa_eth: make the common code endian portable

The parse results structure is in the main memory and is used by the FMan, which
is BE. Also, the release function for an SG table entry should use the address
in a proper CPU endianness.

Signed-off-by: Marian-Cristian Rotariu <marian.rotariu@freescale.com>
Change-Id: Iddc6ee051426a1a584e3a8b15a18b6b1544d8b49
Reviewed-on: http://git.am.freescale.net:8181/22451
Reviewed-by: Madalin-Cristian Bucur <madalin.bucur@freescale.com>
Reviewed-on: http://git.am.freescale.net:8181/32500
[Yang: Original patch taken from
QorIQ-SDK-V1.8-SOURCE-20150619-yocto.iso]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
index be4c317..973eb16 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
@@ -1548,7 +1548,7 @@ void dpa_release_sgt(struct qm_sg_entry *sgt)
 			DPA_BUG_ON(sgt[i].extension);
 
 			bmb[j].hi       = sgt[i].addr_hi;
-			bmb[j].lo       = sgt[i].addr_lo;
+			bmb[j].lo       = be32_to_cpu(sgt[i].addr_lo);
 
 			j++; i++;
 		} while (j < ARRAY_SIZE(bmb) &&
@@ -1663,13 +1663,13 @@ int dpa_enable_tx_csum(struct dpa_priv_s *priv,
 	 */
 	switch (ethertype) {
 	case ETH_P_IP:
-		parse_result->l3r = FM_L3_PARSE_RESULT_IPV4;
+		parse_result->l3r = cpu_to_be16(FM_L3_PARSE_RESULT_IPV4);
 		iph = ip_hdr(skb);
 		DPA_BUG_ON(iph == NULL);
 		l4_proto = iph->protocol;
 		break;
 	case ETH_P_IPV6:
-		parse_result->l3r = FM_L3_PARSE_RESULT_IPV6;
+		parse_result->l3r = cpu_to_be16(FM_L3_PARSE_RESULT_IPV6);
 		ipv6h = ipv6_hdr(skb);
 		DPA_BUG_ON(ipv6h == NULL);
 		l4_proto = ipv6h->nexthdr;
-- 
1.9.1

